src/BatteryStateOfChargeService.cpp

```diff
@@ -24,40 +24,65 @@
  *
  *  For further contact, email <software@calgarysolarcar.ca>
  */
-
+#include<QDebug>
+#include "BatteryData.h"
 #include "BatteryStateOfChargeService.h"
-
+#include <QTime>
+#include <math.h>
 namespace
 {
     const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
+    const int SECONDS_TO_HOURS = 3600;
+    const int SECONDS_TO_MINUTUES = 60;
+
 }
 
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
+
 }
 
 BatteryStateOfChargeService::~BatteryStateOfChargeService()
 {
+
 }
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalcurrent_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (newcurrent_ <= 0)
+    {   return true;}
+    else
+    {   return false;}
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    return estimatetime_;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
-{
-    Q_UNUSED(batteryData);
-    // Update your variables here.
+{   
+    newcurrent_ = batteryData.current;
+    double oldcurrent = 0;
+    QTime newtime = batteryData.time;
+    QTime oldtime = newtime;
+    double s1 = QTime(0,0,0).secsTo(newtime);
+    double s2 = QTime(0,0,0).secsTo(oldtime);
+    int timeused = s1 - s2;
+    totalcurrent_ = newcurrent_ - oldcurrent;
```

Comment:
The `oldcurrent` variable is still always 0 at this point regardless of when you update it. 
`totalcurrent_ = newcurrent_ - oldcurrent;` will always be `totalcurrent_ = new current_ - 0;` which isn't really all that useful :P